Storage node crash in staging-beta

FYI: this occurred after scaling down from two to one storage node.

Logs from storage node:
```
{"caller":"node.go:391","level":"debug","module":"worker/storage/committee","msg":"incoming block","round":13525,"runtime_id":"0000000000000000000000000000000000000000000000000000000000000000","ts":"2019-07-27T00:39:32.486279253Z"}
{"caller":"node.go:408","current_round":13525,"err":"roothash: block not found","level":"error","module":"worker/storage/committee","msg":"can't get block for round","round":0,"runtime_id":"0000000000000000000000000000000000000000000000000000000000000000","ts":"2019-07-27T00:39:32.486333161Z"}
panic: can't get block in storage worker

goroutine 765 [running]:
github.com/oasislabs/ekiden/go/worker/storage/committee.(*Node).worker(0xc00abf6780)
	/workdir/go/worker/storage/committee/node.go:409 +0x1a3a
created by github.com/oasislabs/ekiden/go/worker/storage/committee.(*Node).Start
	/workdir/go/worker/storage/committee/node.go:196 +0x3f
```

--- a/go/worker/storage/committee/node.go
+++ b/go/worker/storage/committee/node.go
@@ -314,7 +314,7 @@ func (n *Node) fetchDiff(round uint64, prevRoot *urkelNode.Root, thisRoot *urkel
 }
 
 type inFlight struct {
-	outstanding map[hash.Hash]struct{}
+	outstanding int
 }
 
 func (n *Node) worker() {
@@ -360,8 +360,9 @@ mainLoop:
 				}
 			}
 
-			delete(syncingRounds[lastDiff.round].outstanding, lastDiff.thisRoot.Hash)
-			if len(syncingRounds[lastDiff.round].outstanding) == 0 {
+			// Check if we have synced the given round.
+			syncingRounds[lastDiff.round].outstanding--
+			if syncingRounds[lastDiff.round].outstanding == 0 {
 				delete(syncingRounds, lastDiff.round)
 				summary := hashCache[lastDiff.round]
 				delete(hashCache, lastDiff.round-1)
@@ -419,13 +420,10 @@ mainLoop:
 					continue
 				}
 
-				newSync := &inFlight{
-					outstanding: make(map[hash.Hash]struct{}),
+				syncingRounds[i] = &inFlight{
+					// We are syncing two roots, I/O root and state root.
+					outstanding: 2,
 				}
-				hashes := hashCache[i]
-				newSync.outstanding[hashes.ioRoot.Hash] = struct{}{}
-				newSync.outstanding[hashes.stateRoot.Hash] = struct{}{}
-				syncingRounds[i] = newSync
 
 				prev := hashCache[i-1] // Closures take refs, so they need new variables here.
 				this := hashCache[i]

